PsfLauncher/main.cpp

```diff
@@ -230,6 +244,170 @@ void GetAndLaunchMonitor(const psf::json_object& monitor, std::filesystem::path
     LaunchMonitorInBackground(packageRoot, monitorExecutable->as_string().wide(), monitorArguments->as_string().wide(), wait, asAdmin, cmdShow, dirStr);
 }
 
+void LaunchInBackgroundAsAdmin(const wchar_t executable[], const wchar_t arguments[], bool wait, int cmdShow, LPCWSTR dirStr)
+{
+    std::wstring cmd = L"\"";
+    cmd.append(executable);
+    cmd.append(L"\"");
+
+
+    // This happens when the program is requested for elevation.  For now just launch this way.  We don't get to inject,
+    // but it is better than nothing.
+    SHELLEXECUTEINFOW shExInfo =
+    {
+        sizeof(shExInfo) // bSize
+        , wait ? (ULONG)SEE_MASK_NOCLOSEPROCESS : (ULONG)(SEE_MASK_NOCLOSEPROCESS | SEE_MASK_WAITFORINPUTIDLE) // fmask
+        , 0           // hwnd
+        , L"runas"    // lpVerb
+        , cmd.c_str() // lpFile
+        , arguments   // lpParameters
+        , dirStr     // lpDirectory
+        , cmdShow           // nShow
+        , 0           // hInstApp
+    };
+    Log("PsfLaunch: Unable to launch with injection (possibly due to elevation).  Launch using shell launch without injection.");
+    THROW_LAST_ERROR_IF_MSG(!ShellExecuteEx(&shExInfo), "Error starting monitor using ShellExecuteEx");
+    THROW_HR_IF(HRESULT_FROM_WIN32(ERROR_INVALID_HANDLE), shExInfo.hProcess == INVALID_HANDLE_VALUE);
+
+    // possibly inject the runtime now?
+    try
+    {
+        BOOL b;
+        BOOL res = IsProcessInJob(shExInfo.hProcess, nullptr, &b);
+        if (res != 0)
+        {
+            if (b == TRUE)
+            {
+                Log("\tPsfLaunch: New process is in a job, allow injection.");
+                // Fix for issue #167: allow subprocess to be a different bitness than this process.
+                USHORT bitness = ProcessBitness(shExInfo.hProcess);
+                DWORD procId = GetProcessId(shExInfo.hProcess);
+                std::filesystem::path packageRoot = PSFQueryPackageRootPath();
+#if _DEBUG

```

Comment:
Why `if DEBUG`? why not always log?

Reply:
To ensure the log functions are only active during debugging, encapsulate them within a preprocessor directive like #ifdef DEBUG. This guards the logging code, activating it only when the debug flag is set, ensuring it's utilized solely for debugging purposes.

Reply:
I apologize.  You interpreted my question in a different way than I was expecting it to be.  I understand what happens in code in a `#ifdef DEBUG` what I don't understand is why include `#IFDEF DEBUG` at all?  Why not have this be logged all the time?  It should help diagnose issues in development.

Reply:
Removed the debug tag in the latest commit. Thanks for the suggestion.